docs/concepts/resource-classes.md

```diff
@@ -0,0 +1,181 @@
+# Kong Ingress Controller and resource class
+
+
+## Table of Contents
+
+- [**Introduction**](#introduction)
+- [**Configuring the controller class**](#configuring-the-controller-class)
+- [**Kubernetes resource class**](#kubernetes-resource-class)
+- [**When to use custom classes**](#when-to-use-custom-classes)
+- [**Legacy behavior**](#legacy-behavior)
+- [**Examples**](#examples)
+
+## Introduction
+
+The Kong Ingress Controller uses resource classes to filter Kubernetes Ingress
+objects and other resources before converting them into Kong configuration.
+This allows it to coexist with other ingress controllers and/or other
+deployments of the Kong Ingress Controller in the same cluster: a Kong Ingress
+Controller will only process configuration marked for its use.
+
+## Configuring the controller class
+
+The `--ingress-class` flag (or `CONTROLLER_INGRESS_CLASS` environment variable)
+specify the resource class expected by the Kong Ingress Controller. By default,
+it expects the `kong` class.
+
+## Kubernetes resource class
+
+The Kong Ingress Controller translates a variety of Kubernetes resources into
+Kong configuration. Broadly speaking, we can separate these resources into two
+categories:
+
+- Resources that the controller translates directly into Kong configuration.
+- Resources referenced by some other resource, where the other resource is
+  directly translated into Kong configuration.
+
+For example, an Ingress is translated directly into a Kong route, and a
+KongConsumer is translated directly into a Kong consumer. A Secret containing
+an authentication plugin credential is _not_ translated directly: it is only
+translated into Kong configuration if a KongConsumer resource references it.
+
+Because they create Kong configuration indenpendent of any other resources,
+directly-translated resources require a class, and their class must match the
+class configured for the controller. Referenced resources do not require a
+class, but must be referenced by a directly-translated resource matches the
+controller.
+
+### Adding class information to resources
+
+Most resources use a [kubernetes.io/ingress-class annoration][class-annotation]
+to indicate their class. There are several exceptions:
+
+- v1 Ingress resources have a dedicated `class` field.
+- Knative Services [use the class specifed][knative-class] by the
+  `ingress.class` key of the Knative installation's `config-network` ConfigMap.
+  You can optionally [override this on a per-Service basis][knative-override]
+  by adding a `networking.knative.dev/ingress.class` annotation to the Service.
+
+### Allowing resources without a class
+
+Specifying a class is optional for some resources. Although specifying a class
+is recommended, you can instruct the controller to process resources without a
+class annotation using flags:
```

Comment:
This sounds very complicated because of the passive voice.

I suggest something along those lines:

For many resource kinds <link to table>, Kong excludes from processing all objects of those kinds which do not have a matching ingress class set. For certain resource type <url or list>, you can instruct Kong to include objects with no class by setting the following flags:

Reply:
Overruled; it isn't; diatribe explanation elsewhere.

Reply:
Please at least make a short explanation as to _why_ specifying a class is recommended, or link to such. Also, please be specific, for _which_ resources specifying a class is not optional. The table mentioned in my other comment could help with that.

Reply:
Such explanation follows within 4 lines. Objection noted and dismissed.

Reply:
> Please at least make a short explanation as to **why specifying a class is recommended** (...)

I don't see how this was fixed.